### ISSUE TITLE
yolov26 overlapped boxes

### ISSUE THREAD

**Issue by guetaro** [ultralytics/ultralytics#23850]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

Hi, I'm using yolov26 model.val function.
In my results there are a lot of extremally overlapped bboxes.
I know the v26 is NMS free version, and that the model should learn to provide one bbox per object, but even after a lot of epochs (more then 100), it still provides those results.
Any one faced this issue? any ideas how to fix it?
Thanks  

### Environment

python 3.10, Pycharm.

### Minimal Reproducible Example

model.val

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @guetaro, thank you for reporting this and for using Ultralytics 🚀 This is an automated response to help get the issue moving quickly, and an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many common questions are already covered.

Since this appears to be a 🐛 Bug Report, please provide a true [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can investigate properly. `model.val` alone is not enough to reproduce the behavior. Please include:
-Your validation code snippet
-Model name/checkpoint used
-Any relevant `conf`, `iou`, `imgsz`, and validation settings
-A small sample image or dataset subset if possible
-The exact console output/logs
-Your installed `ultralytics` version

If this is instead a custom training ❓ question, please also share as much detail as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community 💬

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments, with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled:
-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by onuralpszr** [comment]
@guetaro model.val using  conf=0.001 so you need to use normal predict see normal results.

**Comment by guetaro** [comment]
Hi, @onuralpszr , thanks for your answer, I'm using conf=0.2 , so that's not the problem.

**Comment by Y-T-G** [comment]
@guetaro Are the overlapping boxes same class or different classes?

**Comment by guetaro** [comment]
Hi @Y-T-G , the case is of one class only, so the same class.

**Comment by Y-T-G** [comment]
How many images do you have for training?

**Comment by guetaro** [comment]
Tens of thousands, I don't believe it's a problem of limited data.

**Comment by Y-T-G** [comment]
You can use NMS-based inference if NMS-free is not working adequately for you. You can run `model.val()` with `end2end=False`

**Comment by guetaro** [comment]
Well, it works, the duplicated bboxes removed - thank you!
But now I facing another issue - it seems that even the gt and pred boxes are with higher IOU than threshold, sometimes it considers it as FP + FN (instead of just TP). Again, there is one label only. In this case the IOU threshold was 0.1 but bboxes with higher IOU raised this problem.
TY again

**Comment by Y-T-G** [comment]
The `iou` argument doesn't control the IoU used for matching predictions with ground truth. It contrils IoU used by NMS. `iou` for matching is fixed at 0.45

**Comment by guetaro** [comment]
I see. So how can I control that param? using argument or even in the code itself.
TY again.

**Comment by Y-T-G** [comment]
It's part of the code here:

https://github.com/ultralytics/ultralytics/blob/f55a0ccd79312703b0e1e2d9f38aecab3a6dd202/ultralytics/utils/metrics.py#L374

**Comment by BillionClaw** [comment]
Looking into the YOLOv26 overlapped boxes issue. Will submit a fix.

**Comment by BillionClaw** [comment]
After looking into this further, this appears to be expected behavior for the NMS-free architecture rather than a code bug — the workaround with `end2end=False` is the right approach for cases where the model hasn't fully converged on duplicate suppression. No code fix needed here.

**Comment by glenn-jocher** [comment]
Thanks for checking into this — since `end2end=False` resolves the duplicates, this looks like expected YOLO26 end-to-end behavior rather than a code bug. For the TP/FP/FN mismatch, `val()` does not currently expose the GT/pred matching IoU as an argument: the confusion-matrix path is fixed in [`ConfusionMatrix.process_batch()`](https://docs.ultralytics.com/reference/utils/metrics/#ultralytics.utils.metrics.ConfusionMatrix.process_batch), while mAP matching is handled separately in [`DetectionValidator._process_batch()`](https://docs.ultralytics.com/reference/models/yolo/detect/val/#ultralytics.models.yolo.detect.val.DetectionValidator._process_batch) and [`match_predictions()`](https://docs.ultralytics.com/reference/engine/validator/#ultralytics.engine.validator.BaseValidator.match_predictions).
